### PR TITLE
feat(v1): add ELK auto-layout for canonical imports

### LINKED ISSUE #1211: (GSoC'26) Feature: Auto-layout circuits without a layout format using ELK.js
### Is your feature request related to a problem? Please describe

The circuits with proper net and connectivity information can't be constructed into the canvas if the layout is not provided. This features solve the problem by generating layout for the circuits whose json contains correct netlist and nets of canonical pipeline.

### Describe the solution you'd like

Automatically apply an ELK.js-based layout to circuits that do not have a layout format. The layout should arrange circuit components and connections in a clear, readable manner without requiring manual positioning.

### Describe alternatives you've considered

_No response_

### Is requested feature related to vue simulator?

v1

### Additional context

_No response_

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by Harkeerat24**
Fixes #1211

#### Describe the changes you have made in this PR -

- Added `elkjs` as a project dependency and dynamically load the auto-layout module only when layout generation is required.
- Added ELK.js-based automatic layout generation for canonical circuits that do not contain layout data.
- Added a worker-based ELK layered layout with left-to-right component placement and orthogonal wire routing.
- Preserved existing canonical layouts by only running auto-layout when the imported circuit has no layout.
- Used actual component dimensions, directions, and fixed port positions while constructing the ELK graph.
- Constrained `Input` components to the first layer and `Output` components to the last layer for a more natural circuit flow.
- Added synthetic zero-size junction nodes for fan-out nets while keeping every ELK edge as a one-source to one-target connection.
- Converted ELK bend points and junction positions back into canonical intermediate routing nodes and connections.
- Deduplicated generated routing nodes and connections and snapped generated coordinates to the simulator grid.

### Screenshots of the UI changes (If any) -

https://github.com/user-attachments/assets/6752888b-8bf0-472c-913a-b196927da0ae

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance

**Explain your implementation approach:**

- When canonical layout data is missing, components are first constructed without depending on saved coordinates so their dimensions and ports are available for layout generation.
- I convert the circuit into an ELK graph using the actual component dimensions and port positions. ELK then uses its layered algorithm with orthogonal routing to calculate component positions and wire paths.
- For two-terminal nets, a normal source-to-target ELK edge is used. For fan-out nets, I use a temporary zero-size junction with a source-to-junction and junction-to-target branches so the shared net topology is represented explicitly during layout.
- After ELK finishes, the generated component positions, bend points, and junction positions are converted back into the canonical layout format.
- Generated coordinates are snapped to the simulator grid, while duplicate routing nodes and connections are avoided.
- The ELK-specific junction representation exists only inside the auto-layout adapter and is converted back into the existing canonical routing representation before the normal import flow continues.
- Circuits that already contain layout data continue using their saved layout and do not run ELK auto-layout.

---

## Checklist before requesting a review

- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

## Summary by CodeRabbit

* **New Features**
  * Added automatic circuit layout generation when layout data is unavailable.
  * Improved routing for complex connections, including multi-target nets and intermediate junctions.
  * Added grid-aligned component and connection positioning.
  * Added orthogonal routing with automatic bends for cleaner circuit diagrams.
* **Bug Fixes**
  * Improved circuit loading when component layout information is missing by applying safe default positions.
  * Preserved component labels, metadata, port positions, and scope information during automatic layout.

### CODE REVIEW

File: v1/src/simulator/src/data/autoLayout.ts [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+    }
+
+    const bounds = getBounds(instance);
+
+    layout[child.id] = {
+      x: snap(child.x + bounds.left),
+      y: snap(child.y + bounds.up),
+      labelDirection: instance.labelDirection,
+    };
+  }
```
[the remark this answers is not in the document]
Comment (Harkeerat24) [reply v1/src/simulator/src/data/autoLayout.ts:635]: @coderabbitai Fixed by applying each component’s snap delta to its adjacent bend points and adding bends when a straight ELK edge becomes misaligned after snapping.

**Comment by Harkeerat24** [comment]
@coderabbitai review

**Comment by Harkeerat24** [comment]
@aryanndwi123 @JoshVarga @tachyons PTAL:)
